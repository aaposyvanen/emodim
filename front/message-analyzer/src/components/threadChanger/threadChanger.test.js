import React from "react";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import ThreadChanger from "./threadChanger";

test("ThreadChanger renders correctly", () => {
    const mockUpdateIndex = jest.fn();

    const { container } = render(
        <ThreadChanger
            currentIndex={0}
            updateIndex={mockUpdateIndex}
            maxIndex={1}
        />
    );
    expect(container).toMatchSnapshot();
});

test("ThreadChanger calls updateIndex on decrease", () => {
    const mockUpdateIndex = jest.fn();

    render(<ThreadChanger
        currentIndex={2}
        updateIndex={mockUpdateIndex}
        maxIndex={5}
    />);

    userEvent.click(screen.getByText("Prev"));
    expect(mockUpdateIndex).toHaveBeenCalledTimes(1);
    expect(mockUpdateIndex).toHaveBeenCalledWith(1);
});

test("ThreadChanger calls updateIndex on increase", () => {
    const mockUpdateIndex = jest.fn();

    render(<ThreadChanger
        currentIndex={1}
        updateIndex={mockUpdateIndex}
        maxIndex={2}
    />);

    userEvent.click(screen.getByText("Next"));
    expect(mockUpdateIndex).toHaveBeenCalledTimes(1);
    expect(mockUpdateIndex).toHaveBeenCalledWith(2);
});

test("ThreadChanger doesn't decrease index under 0", () => {
    const mockUpdateIndex = jest.fn();

    render(<ThreadChanger
        currentIndex={0}
        updateIndex={mockUpdateIndex}
        maxIndex={1}
    />);

    userEvent.click(screen.getByText("Prev"));
    expect(mockUpdateIndex).toHaveBeenCalledTimes(0);
});

test("ThreadChanger doesn't increase index above max", () => {
    const mockUpdateIndex = jest.fn();

    render(<ThreadChanger
        currentIndex={1}
        updateIndex={mockUpdateIndex}
        maxIndex={1}
    />);

    userEvent.click(screen.getByText("Next"));
    expect(mockUpdateIndex).toHaveBeenCalledTimes(0);
});
