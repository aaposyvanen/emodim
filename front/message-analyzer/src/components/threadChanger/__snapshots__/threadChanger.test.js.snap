// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadChanger renders correctly 1`] = `
<div>
  <div
    class="thread-changer"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Prev
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="current-thread-label"
    >
      1
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Next
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
